File: yangxj96-starter/yangxj96-starter-remote/src/main/java/io/github/yangxj96/starter/remote/fallback/RemoteProperties.java
package io.github.yangxj96.starter.remote.fallback;

import feign.Logger;
import feign.Retryer;
import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

@Data
@ConfigurationProperties(prefix = "it100000.remote")
public class RemoteProperties {

	/** 连接超时时间 **/
	private Integer connectTimeOut = 1000;

	/** 读取超时时间 **/
	private Integer readTimeOut = 1000;

	/** 写出超时时间 **/
	private Integer writeTimeout = 1000;

	/** feign日志输出等级 **/
	private Logger.Level level = Logger.Level.FULL;

	/** feign 重试配置 **/
	private feign.Retryer retryer = new Retryer.Default();



}

File: yangxj96-starter/yangxj96-starter-remote/src/main/java/com/yangxj96/saas/starter/remote/props/RemoteProperties.kt
/*
 * Copyright (c) 2018 - 2023
 * 作者：杨新杰(Jack Young)
 * 邮箱：yangxj96@gmail.com
 * 博客：www.yangxj96.com
 * 日期：2023-09-13 00:04:11
 * Copyright (c) 2018 - 2023
 */

package com.yangxj96.saas.starter.remote.props

import feign.Logger
import feign.Retryer
import org.springframework.boot.context.properties.ConfigurationProperties
import java.util.concurrent.TimeUnit

/**
 * 远程相关配置
 */
@ConfigurationProperties(prefix = "yangxj96.remote")
class RemoteProperties {
    /**
     * 连接超时时间,单位毫秒.
     */
    var connectTimeOut = 1000L

    /**
     * 读取超时时间,单位毫秒.
     */
    var readTimeOut = 1000L

    /**
     * 写出超时时间,单位毫秒.
     */
    var writeTimeout = 1000L

    /**
     * feign日志输出等级
     */
    var level = Logger.Level.NONE

    /**
     * feign 重试配置
     */
    var retryer: Retryer = Retryer.Default(100, TimeUnit.SECONDS.toMillis(1), 2)
}
